Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -340,12 +341,25 @@ public ActionResult Retest(int id)
             }
             else
```

Comment:
I think adding a redirect into the if (submission == null) itself might be a good idea, because the need for the else will disappear making the code more readable.